src/components/molecules/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,59 @@
+import VenueChat from "components/molecules/VenueChat";
+import ChatsList from "components/molecules/ChatsList";
+import LiveSchedule from "components/molecules/LiveSchedule";
+import { useSelector } from "hooks/useSelector";
+import { useVenueId } from "hooks/useVenueId";
+import React, { useState } from "react";
+import { useFirestoreConnect } from "react-redux-firebase";
+import "./Sidebar.scss";
+
+const Sidebar = () => {
+  const venueId = useVenueId();
+  useFirestoreConnect({
+    collection: "users",
+    where: ["enteredVenueIds", "array-contains", venueId],
+    storeAs: "chatUsers",
+  });
+  const [tab, setTab] = useState(0);
+  const { privateChats, chatUsers } = useSelector((state) => ({
+    privateChats: state.firestore.ordered.privatechats,
+    chatUsers: state.firestore.data.chatUsers,
+  }));
+  const isEnabled = chatUsers && privateChats;
+  return (
+    <div className="sidebar-container">
+      <div className="sidebar-slide-btn">
+        <div className="slide-btn-arrow-icon"></div>
+        <div className="slide-btn-chat-icon"></div>
+      </div>
+      <div className="sidebar-tabs">
+        <div
+          className={`sidebar-tab sidebar-tab_chat ${tab === 0 && "active"}`}
+          onClick={() => isEnabled && setTab(0)}
+        >
+          Party Chat
+        </div>
+        <div
+          className={`sidebar-tab sidebar-tab_private ${tab === 1 && "active"}`}
+          onClick={() => isEnabled && setTab(1)}
+        >
+          Messages
+        </div>
+        <div
+          className={`sidebar-tab sidebar-tab_schedule ${
+            tab === 2 && "active"
+          }`}
+          onClick={() => isEnabled && setTab(2)}
+        >
+          Live Schedule
+        </div>
+      </div>
+
+      {tab === 0 && <VenueChat />}
+      {tab === 1 && <ChatsList />}
+      {tab === 2 && <LiveSchedule />}
```

Comment:
[nit] I wonder if we can extract these tab indexes in some way that makes them more semantically obvious/useful? Currently I would need to go lookup the 'meaning' of each `tab` value to be able to assess if it's behaviour is correct or not.

Reply:
An alternative (probably more efficient) solution to the above comments with individual `useCallback`'s, that would also solve this nit:

Inside the component:


```typescript
const [tab, setTab] = useState(0);

const isActiveTab = (_tab) => _tab === tab

const {
  selectVenueChat,
  selectChatsList,
  selectLiveSchedule,
  renderTab,
} = useMemo(() => {  
  const tabs: {
     _default: VenueChat,
    venueChat: VenueChat,
    chatsList: ChatsList,
    liveSchedule: LiveSchedule,
  }

  const selectTab = (_tab) => {
    if (isEnabled) {
      setTab(tabs[_tab])
    }
  }

  return {
    selectVenueChat: () => selectTab('venueChat'),
    selectChatsList: () => selectTab('chatsList'),
    selectLiveSchedule: () => selectTab('liveSchedule'),
    renderTab => (_tab) => tabs[_tab] ?? <>Unknown tab selected</>
}, [isEnabled])
```

For each of the tabs, you'll now have a memo'd function that can be used directly, eg:

```typescript
        <div
          className={`sidebar-tab sidebar-tab_chat isActiveTab('partyChat') && "active"}`}
          onClick={selectPartyChat}
        >
          Party Chat
        </div>
```

Then here you would just use:
```suggestion
      {renderSelectedTab(tab)}
```

Reply:
Let's do the above, and also @denisdimitrov let's use an enum for these tab identifiers rather than a number.

Reply:
fixed in 70e8253242c4c56f3678a5035e0e6566912da044

@denisdimitrov kept the 3 lines in the JSX, but using the enum as you suggested. It looks a lot simpler and cleaner than my above suggestion the way it is done now, so I say :shipit: 🎉